ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/KsqlServerMain.java

```diff
@@ -101,6 +102,47 @@ void tryStartApp() throws Exception {
     }
   }
 
+  private static void validateConfig(final KsqlConfig config) {
+    validateStateDir(config);
+    validateDefaultTopicFormats(config);
+  }
+
+  private static void validateStateDir(final KsqlConfig config) {
+    final String streamsStateDirPath = config.getKsqlStreamConfigProps().getOrDefault(
+        StreamsConfig.STATE_DIR_CONFIG,
+        StreamsConfig.configDef().defaultValues().get(StreamsConfig.STATE_DIR_CONFIG)).toString();
+    enforceStreamStateDirAvailability(new File(streamsStateDirPath));
+  }
+
+  @VisibleForTesting
+  static void validateDefaultTopicFormats(final KsqlConfig config) {
+    validateTopicFormat(config, KsqlConfig.KSQL_DEFAULT_KEY_FORMAT_CONFIG, "key");
+    validateTopicFormat(config, KsqlConfig.KSQL_DEFAULT_VALUE_FORMAT_CONFIG, "value");
+  }
+
+  private static void validateTopicFormat(
+      final KsqlConfig config,
+      final String configName,
+      final String type
+  ) {
+    final String formatName = config.getString(configName);
+    if (formatName == null) {
+      return;
+    }
+
+    final Format format;
+    try {
+      format = FormatFactory.fromName(formatName.toUpperCase());
```

Comment:
You upper-case here, but not elsewhere in the code when accessing the property.  Maybe just move the uppercasing into `fromName`?